### DISCUSSION TITLE
Automated Project Creation?

### DISCUSSION THREAD

**Discussion by gthomsen** [scalabel/scalabel#409] [Q&A]
Is it possible to automate the creation of new labeling projects?

I've built tools to prepare my data and create the items list JSON and categories YAML, and would like to streamline the process of uploading them into Scalabel.  My data and Scalabel live on the system that generates the configuration files, though the step to transfer configurations to my local system just so I can create a project and label is error prone and seems unnecessary.

I'm happy to write the automation tools, I just need guidance on the interface or where to look in the code.

Thanks!

**Comment by fyu** [comment]
It would be great to have your contribution to the automated project creation. You can look at the nodejs code for the project creation. It is possible to reuse the creation javascript code and have a nodejs command line code to generate the same content.

**Reply by gthomsen** [reply]
I'll take a look at the nodejs command line approach, though I've never created a node-based tool before.  

Would it be possible to post the required configurations via a HTTP request so the tool can run outside of the container?  If so, where is a good starting point to look?

**Reply by blakete** [reply]
Great question. Looking forward to hearing back on this.
